structuralcodes/sections/specific.py

```diff
@@ -0,0 +1,446 @@
+""" Specific class section implementation"""
+
+from __future__ import annotations
+
+import typing as t
+import structuralcodes.core._section_results as s_res
+from shapely import Polygon
+from structuralcodes.sections._generic import GenericSection
+from structuralcodes.geometry import SurfaceGeometry
+from structuralcodes.materials.concrete import Concrete
+
+class RectangularSection(GenericSection): 
+    """This is the specific implementation of a rectangular class section.
+
+    The section is a 2D geometry where Y axis is horizontal while Z axis is
+    vertical.
+
+    The moments and curvatures around Y and Z axes are assumed positive
+    according to RHR.
+
+    Attributes: TODO
+        geometry (Union(SurfaceGeometry, CompoundGeometry)): The geometry of
+            the section.
+        name (str): The name of the section.
+        section_calculator (GenericSectionCalculator): The object responsible
+            for performing different calculations on the section (e.g. bending
+            strength, moment curvature, etc.).
+    """
+
+    def __init__(self, height: float, width: float, concrete: Concrete, name: t.Optional[str] = None, integrator: t.Literal['marin', 'fiber'] = 'marin') -> None:
+        """Initialize a GenericSection.
+
+        Arguments:
+            geometry (Union(SurfaceGeometry, CompoundGeometry)): The geometry
+                of the section.
+            name (str): The name of the section.
+            integrator (str): The name of the SectionIntegrator to use.
+        """
+        poly: Polygon = Polygon(((0, 0), (width, 0), (width, height), (0, height)))
+        geometry: SurfaceGeometry = SurfaceGeometry(poly, concrete)
+        geometry.translate(-width/2, -height/2)
+        if name is None:
+            name: str = 'RectangularSection'
+        super().__init__(geometry, name, integrator)
+
+        self.height: float = height
+        self.width: float = width
+
+        # Calculating gross section properties
+        # Gross area
+        area: float = height * width
+        perimeter: float = 2 * (height + width)
+        ea: float = area * geometry.material.get_tangent(eps=0)[0]
+        for geo in self.geometry.geometries:
+            pass
+        for geo in self.geometry.point_geometries:
+            pass
+
+        if geometry.density is not None:
+            mass = area * geometry.density * 1e-9
+        else:
+            mass = 0
```

Comment:
Do we need to compute this stuff? Or is it already tackled by `GenericSection`?

Reply:
I believe you are correct. How do we want to proceed? The idea behind the specific section is, as I understand it, to make it more efficient. Thus, calculating gross properties directly analytically would be the way to go?

Reply:
I agree with your comment. I suggest to go one step at a time. As a first step let's just create a wrapper (no optimization intended). When this is working we will include the efficient computations.

In particular if we want to rewrite the `calculate_gross_section_properties` method we should compute all the values computed in the generic section. There is way more than just mass, area, ea.

Moreover I am not following here the code: why you compute area_reinforcement if this is a plain rectangular section?